modules/consensus/consensusset.go

```diff
@@ -222,6 +228,12 @@ func (cs *ConsensusSet) CurrentBlock() (block types.Block) {
 	return block
 }
 
+// Flush will block until the consensus set has finished all in-progress
+// routines.
+func (cs *ConsensusSet) Flush() error {
+	return cs.tg.Flush()
+}
```

Comment:
I feel like we shouldn't need this level of introspection in order to do proper integration testing. This creates some pretty tight coupling.